lib/module.js

```diff
@@ -38,37 +42,43 @@ module.exports = function sentry(moduleOptions) {
     return
   }
 
-  // Initialize Sentry
-  Sentry.init({
-    dsn: options.dsn,
-    ...options.config
-  })
-  logger.success('Started logging errors to Sentry')
-
-  // Enable publishing of sourcemaps
-  this.extendBuild((config, { isClient, isDev }) => {
-    if (!options.publishRelease || isDev) return
-    if (isClient) config.devtool = '#sourcemap'
-
-    config.plugins.push(new WebpackPlugin(options.webpackConfig))
-    logger.info('Enabling uploading of release sourcemaps to Sentry')
-  })
-
   // Register the client plugin
   if (!options.disableClientSide) {
     this.addPlugin({
       src: path.resolve(__dirname, 'sentry.client.js'),
       fileName: 'sentry.client.js',
-      ssr: false,
+      mode: 'client',
       options: {
         dsn: options.dsn,
         ...options.config
       }
     })
   }
 
-  // Initialize the hooks
-  if (this.nuxt.hook) {
+  // Register the server plugin
+  if (!options.disableServerSide) {
+    // Initialize Sentry
+    Sentry.init({
```

Comment:
I think either this or `sentry.server.js` is duplicate as both are running in the same porcces. What do you think to Init sentry here and inject it to `process.sentry` so the server plugin can inject it directly as `$sentry`?

Reply:
I agree that Init is duplicated here and in `sentry.server.js`, but I I'm not sure what would be advantage of Initializing here and passing through `process.sentry`. Wouldn't it make sense to initialize in `sentry.server.js`, just not inside the exported function, but in global scope of the plugin, so that Sentry is not re-initialized on each request?

Reply:
Now I understand... We need to pass options and can't initialize before that. So your solution sounds good to me.